src/test/scala/com/databricks/spark/avro/AvroSuite.scala

```diff
@@ -297,6 +298,25 @@ class AvroSuite extends FunSuite with BeforeAndAfterAll {
     }
   }
 
+  test("Date field type") {
+    TestUtils.withTempDir { dir =>
+      val schema = StructType(Seq(
+        StructField("float", FloatType, true),
+        StructField("date", DateType, true)
+      ))
+      val rdd = spark.sparkContext.parallelize(Seq(
+        Row(1f, null),
+        Row(2f, new Date(1451980800000L)),
+        Row(3f, new Date(1460098800000L))
+      ))
+      val df = spark.createDataFrame(rdd, schema)
+      df.write.avro(dir.toString)
+      assert(spark.read.avro(dir.toString).count == rdd.count)
+      assert(spark.read.avro(dir.toString).select("date").collect().map(_(0)).toSet ==
+        Array(null, 1451980800000L, 1460098800000L).toSet)
```

Comment:
Hi, 
The issue is relative to the timestamp, We are managing Date so there is a round done to keep only the day at 00:00:00 in your local time zone 

How I fixed : 
      TimeZone.setDefault(TimeZone.getTimeZone("UTC"))
      val rdd = spark.sparkContext.parallelize(Seq(
        Row(1f, null),
        Row(2f, new Date(1451948400000L)),
        Row(3f, new Date(1460066400500L))
      ))
      val df = spark.createDataFrame(rdd, schema)
      df.write.avro(dir.toString)
      assert(spark.read.avro(dir.toString).count == rdd.count)
      assert(spark.read.avro(dir.toString).select("date").collect().map(_(0)).toSet ==
        Array(null, 1451865600000L, 1459987200000L).toSet)